### PR TITLE
Fix for revoke flow and cleanup

### PR DESCRIPTION
**Pull request by chrisnojima**
This is split into 3 commits
1. we were using call instead of put for navigateTo which broke which screen we went to
1. better layout for login screen. instead of all these fixed margins lets use centering and auto growing boxes to make sure things look centered no matter what's going on
1. general cleanup of files. i had a really hard time looking at the hierarchy since a lot of things wre called 'Render'. This updates the login screens to be container.js index.desktop.js etc. Also splitting the connect functions into helpers like we should everywhere.

### CODE REVIEW

File: shared/actions/devices.js [hunk 1]
```diff
@@ -131,7 +131,7 @@ function* _deviceRevokedSaga(action: Revoke): SagaGenerator<any, any> {
       }
       yield put(setWaiting(true))
       yield call(loginDeprovisionRpcPromise, {param: {doRevoke: true, username}})
-      yield call(navigateTo([loginTab]))
+      yield put(navigateTo([loginTab]))
```
Comment (chrisnojima) [inline shared/actions/devices.js]: the real bug

File: shared/common-adapters/header-hoc.native.js [hunk 2]
```diff
@@ -51,7 +51,6 @@ const _headerStyle = {
   borderBottomColor: globalColors.black_05,
   borderBottomWidth: StyleSheet.hairlineWidth,
   justifyContent: 'flex-start',
-  marginTop: statusBarHeight,
```
Comment (chrisnojima) [inline shared/common-adapters/header-hoc.native.js]: cc: @chromakode this is to fix the double marginTop. we can discuss if this doesn't make sense

File: shared/folders/files/paperkey.js [hunk 3]
```diff
@@ -19,7 +19,8 @@ type Props = {
   checkPaperKey: (paperKey: HiddenString) => void,
 }
 
-class PaperKey extends Component<void, Props, {paperKey: string}> {
+// TODO remove this class
+class _PaperKey extends Component<void, Props, {paperKey: string}> {
```
Comment (chrisnojima) [inline shared/folders/files/paperkey.js]: all of these types of 'helper' classes in connect can likely be replaced by recompose

File: shared/login/forms/index.native.js [hunk 4]
```diff
@@ -101,11 +103,20 @@ const Intro = (props: Props) => (
           Your device has been revoked, please log in again.
         </Text>
       </Box>}
-    <Box style={{...globalStyles.flexBoxColumn, flex: 1, justifyContent: 'center', alignItems: 'center'}}>
+    <Box
```
Comment (chrisnojima) [inline shared/login/forms/index.native.js]: center more things

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready

**Comment by keybase-ci-visdiff** [comment]
:no_entry: Error rendering the commits 58d2086310f0...2b6518162be6 on linux. <details><summary>:mag_right: 4 errored, 1 changed</summary>

 * Error in 2b6518162be6: **Signup_Success-Start-ERROR**  
   Signup_Success-Start-ERROR rendered
 * Error in 2b6518162be6: **Signup_Passphrase-Start-ERROR**  
   Signup_Passphrase-Start-ERROR rendered
 * Error in 2b6518162be6: **Signup_Passphrase-Error-ERROR**  
   Signup_Passphrase-Error-ERROR rendered
 * Error in 2b6518162be6: **Signup_Error-Start-ERROR**  
   Signup_Error-Start-ERROR rendered
 * Changed: **Tracker-New_user_Scroll_1** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1497291523009-58d2086310f0-2b6518162be6-linux/Tracker-New_user_Scroll_1-58d2086310f0.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1497291523009-58d2086310f0-2b6518162be6-linux/Tracker-New_user_Scroll_1-2b6518162be6.png)  
   Tracker-New_user_Scroll_1 comparison
</summary>

**Comment by keybase-ci-visdiff** [comment]
:no_entry: Error rendering the commits 97c6a254ff53...a4a58df6f179 on linux. <details><summary>:mag_right: 4 errored</summary>

 * Error in a4a58df6f179: **Signup_Success-Start-ERROR**  
   Signup_Success-Start-ERROR rendered
 * Error in a4a58df6f179: **Signup_Passphrase-Start-ERROR**  
   Signup_Passphrase-Start-ERROR rendered
 * Error in a4a58df6f179: **Signup_Passphrase-Error-ERROR**  
   Signup_Passphrase-Error-ERROR rendered
 * Error in a4a58df6f179: **Signup_Error-Start-ERROR**  
   Signup_Error-Start-ERROR rendered
</summary>

File: shared/actions/devices.js (hunk 1 above)
Comment (MarcoPolo) [inline shared/actions/devices.js]: This has happened before, right? Should we grep around to make sure this doesn't happen elsewhere

File: shared/main-shared.native.js [hunk 5]
```diff
@@ -73,7 +73,7 @@ class Main extends Component<void, any, void> {
     // TODO: move Push prompt into route
     const {showPushPrompt, mountPush} = this.props
     return (
-      <Box style={{flex: 1}}>
+      <Box style={{flex: 1, width: '100%'}}>
```
Comment (MarcoPolo) [inline shared/main-shared.native.js]: Why is this necessary?

Comment (chrisnojima) [inline shared/main-shared.native.js]: its not strictly necessary but the flex 1 doesn't actually fill the entire parent for some reason, this makes it do so (could see in the debugger)

File: shared/actions/devices.js (hunk 1 above)
Comment (chrisnojima) [inline shared/actions/devices.js]: i'd prefer if flow could catch this :)
